webview-ui/src/components/settings/SettingsView.tsx

```diff
@@ -217,13 +217,16 @@ const SettingsView = forwardRef<SettingsViewRef, SettingsViewProps>(({ onDone },
 	const onConfirmDialogResult = useCallback((confirm: boolean) => {
 		if (confirm) {
 			setChangeDetected(false)
-			// Wait for the change detection to be updated
-			setTimeout(() => {
-				confirmDialogHandler.current?.()
-			}, 100)
 		}
 	}, [])
 
+	useEffect(() => {
+	  if (!isChangeDetected && confirmDialogHandler.current) {
+	    confirmDialogHandler.current();
+	    confirmDialogHandler.current = undefined; // Clear the handler after execution
+	  }
```

Comment:
I don't think this code is right

Reply:
Although this patch works in some way, it’s quite confusing because `isChangeDetected` is tightly coupled with the `confirmDialogHandler` call. 
In reality, it's just a loop of `onDone -> switchTab -> onDone`, and all we need to do is break the loop.
```tsx
{tab === "settings" && <SettingsView ref={settingsRef} onDone={() => setTab("chat")} />}
```

Reply:
Oh that's probably a lot smarter than what I ended up doing in #1090 😬